src/server/modules/user/index.js

```diff
@@ -65,6 +66,60 @@ if (settings.user.auth.facebook.enabled) {
   );
 }
 
+if (settings.user.auth.google.enabled) {
+  passport.use(
+    new GoogleStrategy(
+      {
+        clientID: settings.user.auth.google.clientID,
+        clientSecret: settings.user.auth.google.clientSecret,
+        callbackURL: '/auth/google/callback'
+      },
+      async function(accessToken, refreshToken, profile, cb) {
+        const { id, username, displayName, emails: [{ value }] } = profile;
+        try {
+          let user = await User.getUserByGoogleIdOrEmail(id, value);
+
+          if (!user) {
+            const isActive = true;
+            const [createdUserId] = await User.register({
+              username: username ? username : value,
```

Comment:
I used the email here because I hit the unique constraint on username in the database.

I was logging in with two of my gmail accounts where my display name (`fname + lname`) is the same, but my email is different.

Reply:
@verdverm Yeah, I'm not sure why do we need `username` at all. I think `email` should be enough, for email/password based authentication. @mitjade Mitja maybe you have idea why unique `username` is needed now in `user` module?

Reply:
I would say `email` is for identity and `username` is more of a display name within an application. Users should generally be able to change their display name, and the account is tied to the email.

Facebook requires your username to be your real name, however Reddit does not. It is also possible to tie multiple emails / login providers together into one account in an application by email(s).

I say we leave it for now and deal with it while working on the `rbac` or auth overhaul.